dvclive/__init__.py

```diff
@@ -24,7 +26,7 @@ def init(
     return _metric_logger
 
 
-def log(name: str, val: float, step: int = None):
+def log(name: str, val: Union[int, float], step: int = None) -> None:
```

Comment:
Strictly speaking, you don't need `Union` for `int` with `float`. See [PEP 484](https://www.python.org/dev/peps/pep-0484/#the-numeric-tower). 

> this PEP proposes a straightforward shortcut that is almost as effective: when an argument is annotated as having type float, an argument of type int is acceptable;